test/py/CMakeLists.txt

```diff
@@ -28,6 +28,7 @@ set(VENV_ONNX ${CMAKE_BINARY_DIR}/test/py/venv-onnx)
 set(REQUIREMENTS ${CMAKE_CURRENT_SOURCE_DIR}/requirements.txt)
 set(REQUIREMENTS_ONNX ${CMAKE_CURRENT_SOURCE_DIR}/requirements-onnx.txt)
 set(PYTHON_VERSION_TO_DISABLE_ONNX 3.6)
+option(ENABLE_VIRTUAL_ENV "Enable python virtual environments" ON)
```

Comment:
By default is it On. I thought motivation was to include this flag was to not have it enabled inside CI

Reply:
Thanks for bringing this up Umang, I think its good we make a conscious decision about it.

My initial thoughts were to have it explicitly disabled for CI since that is one scenario, and have it enabled (by default) for all other scenarios unless specified otherwise. If we default to disabling it, all users would have to explicitly enable it to use virtual env.
I am not sure which is a good way to go about. I think it depends on the use cases and how we would like the default to be depending on the use cases? Perhaps @pfultz2 and @causten can weigh in on the preference here.
Thanks!

Reply:
The flag name could also be `DISABLE_VIRTUAL_ENV`, open to feedback on the preference there too.

Reply:
We have control in Jenkins on how to invoke repeatable processes.  The goal of virtual environments is to take ownership of the environment where `make check` is used by others because we have been battling various inconsistencies.  As an example, we moved to onnx==1.10.1 over a year ago, and this week to saw a JIRA showing onnx==1.8.  Now in mainline we have support for "onnx==1.14.1".  

I'd like the default path of `make check` to invoke the virtual environment.  Jenkins scripts can ensure the non virtual environment is done correctly.

Reply:
Okay. we can make it `MIGRAPHX_DISABLE_VIRTUAL_ENV` .  But would need to change Jenkins scripts to pass that flag in this PR.

Reply:
Sounds good. Do you know the location of the Jenkins script that needs to be modified?

Reply:
https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/7604ecf5b9cd04f9df4219d6f1ef998951ed7449/Jenkinsfile#L33 Here you would just need to pass that flag.

Reply:
It would be better to have a stage with virtual env enabled inside Jenkins though to test out Virtual Env as well.

Reply:
> It would be better to have a stage with virtual env enabled inside Jenkins though to test out Virtual Env as well.

I understand the motivation but I dont think its a good idea, we are just adding extra stages that have to download packages already downloaded.

Reply:
> I understand the motivation but I dont think its a good idea, we are just adding extra stages that have to download packages already downloaded. 

Okay in that case we would need to rely on QA for virtualenv testing.  most of us would be using dockers for dev and won't be using virtualenv.